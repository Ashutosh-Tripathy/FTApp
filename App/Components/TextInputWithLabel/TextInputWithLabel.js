import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, Text, TextInput } from 'react-native'

function TextInputWithLabel (props) {
  return (
      <View>
      <Text>{props.labelText}</Text>
      <TextInput multiline {...props} />
      </View>)
}

TextInputWithLabel.propTypes = {
  labelText: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
}

export default TextInputWithLabel
