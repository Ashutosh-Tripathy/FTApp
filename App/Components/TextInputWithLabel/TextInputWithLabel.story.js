import React from 'react'
import { storiesOf } from '@storybook/react-native'

import TextInputWithLabel from './TextInputWithLabel'

let props = {
  labelText: 'Default label',
  value: 'Place holder text',
  onChange: ()=>{}
};

storiesOf('TextInputWithLabel')
.add('Default', () => (
      <TextInputWithLabel {...props} />
      ))
.add('Multiline textinput', () => (
      <TextInputWithLabel multiline={true} {...props } />
      ))
