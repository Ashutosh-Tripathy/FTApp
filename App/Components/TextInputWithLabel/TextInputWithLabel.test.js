import React from 'react';
import { shallow, mount } from 'enzyme';
import TextInputWithLabel from './TextInputWithLabel';
import jsdom from 'jsdom';
import {Text, TextInput} from 'react-native';

describe('MyComponent', () => {
  var props;
  beforeEach(() => {
    props = {
      name: 'Default label',
        value: 'Place holder text',
        onChange: jest.fn()
    };
  })

  it('should render correctly', () => {
    const component = shallow(<TextInputWithLabel {...props} />);
    expect(component).toMatchSnapshot();
  })

  it('should call click event', () => {
    const component = shallow(<TextInputWithLabel {...props} />);
    component.find(TextInput).first().props().onChange(1);
    expect(props.onChange).toHaveBeenCalled();
    expect(props.onChange).toHaveBeenCalledWith(1);
  })
})
