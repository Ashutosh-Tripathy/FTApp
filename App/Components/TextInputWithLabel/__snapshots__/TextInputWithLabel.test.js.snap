// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInputWithLabel
    labelText="Default text"
    onChange={[MockFunction]}
    value="Place holder text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Default text
        </Text>,
        <TextInput
          allowFontScaling={true}
          labelText="Default text"
          multiline={true}
          onChange={[MockFunction]}
          value="Place holder text"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Default text",
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": "Default text",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": true,
          "labelText": "Default text",
          "multiline": true,
          "onChange": [MockFunction],
          "value": "Place holder text",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Default text
          </Text>,
          <TextInput
            allowFontScaling={true}
            labelText="Default text"
            multiline={true}
            onChange={[MockFunction]}
            value="Place holder text"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Default text",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "Default text",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "labelText": "Default text",
            "multiline": true,
            "onChange": [MockFunction],
            "value": "Place holder text",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
